components/scrubber/scrubber.go

```diff
@@ -99,30 +99,37 @@ type Scrubber interface {
 var Default Scrubber = newScrubberImpl()
 
 func newScrubberImpl() *scrubberImpl {
-	hashedRegex, err := regexp.Compile("(?i)(" + strings.Join(HashedFieldNames, "|") + ")")
-	if err != nil {
-		panic(fmt.Errorf("cannot compile hashed regex: %w", err))
+	var (
+		lowerSanitiseHash   []string
+		lowerSanitiseRedact []string
+	)
+	for _, v := range HashedFieldNames {
+		lowerSanitiseHash = append(lowerSanitiseHash, strings.ToLower(v))
+	}
+	for _, v := range RedactedFieldNames {
+		lowerSanitiseRedact = append(lowerSanitiseRedact, strings.ToLower(v))
 	}
 
-	redactedRegex, err := regexp.Compile("(?i)(" + strings.Join(RedactedFieldNames, "|") + ")")
+	cache, err := lru.New(1000)
```

Comment:
I did not look closer. We had this optimisation in first iteration, but then dropped for simplicity. Because was afraid that it can cause memory leaks.

If you tested (ideally with tests) that less frequently records get ejected then let's land.

Reply:
the LRU is a fixed-size LRU, see e.g. the example here: https://github.com/hashicorp/golang-lru#lru-cache-example It will eject keys once going over 1000 entries